### PR TITLE
NAS-132340 / 25.04 / Move reporting exporters service to new api style

### PR DESCRIPTION
**Pull request by Qubad786**
## Context

This PR converts reporting exporter service to new api style.

### CODE REVIEW

File: src/middlewared/middlewared/api/v25_04_0/reporting_exporters.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+
+__all__ = [
+    'ReportingExporterEntry', 'ReportingExporterCreateArgs', 'ReportingExporterCreateResult', 'GraphiteExporter',
+    'ReportingExporterUpdateArgs', 'ReportingExporterUpdateResult', 'ReportingExporterDeleteArgs',
+    'ReportingExporterDeleteResult', 'ReportingExporterSchemasArgs', 'ReportingExporterSchemasResult',
+]
+
+
+### Exporter models
```
Comment (yocalebo) [inline src/middlewared/middlewared/api/v25_04_0/reporting_exporters.py:17]: flake8 should only be 1 #

File: src/middlewared/middlewared/api/v25_04_0/reporting_exporters.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    update_every: int = Field(default=1, ge=1)
+    buffer_on_failures: int = Field(default=10, ge=1)
+    send_names_instead_of_ids: bool = True
+    matching_charts: NonEmptyString = '*'
+
+
+ExporterType: TypeAlias = Annotated[GraphiteExporter, Field(discriminator='exporter_type')]
+
+
+## Reporting Exporter Service
```
Comment (yocalebo) [inline src/middlewared/middlewared/api/v25_04_0/reporting_exporters.py:34]: Same here

File: src/middlewared/middlewared/api/v25_04_0/reporting_exporters.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    'ReportingExporterDeleteResult', 'ReportingExporterSchemasArgs', 'ReportingExporterSchemasResult',
+]
+
+
+### Exporter models
+
+class GraphiteExporter(BaseModel):
+    exporter_type: Literal['GRAPHITE']
+    destination_ip: NonEmptyString
+    destination_port: int = Field(..., ge=1, le=65535)
```
Comment (themylogin) [inline src/middlewared/middlewared/api/v25_04_0/reporting_exporters.py:22]: Why `...`? The default behavior is already the field being required

File: src/middlewared/middlewared/api/v25_04_0/reporting_exporters.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+ExporterType: TypeAlias = Annotated[GraphiteExporter, Field(discriminator='exporter_type')]
+
+
+# Reporting Exporter Service
+
+class ReportingExporterEntry(BaseModel):
+    id: int
+    enabled: bool
+    attributes: ExporterType = Field(..., description='Specific attributes for the exporter')
```
Comment (themylogin) [inline src/middlewared/middlewared/api/v25_04_0/reporting_exporters.py:39]: Same here (`...`)

**Review by themylogin — APPROVED** [review]
Thank you!

**Pull request merged**